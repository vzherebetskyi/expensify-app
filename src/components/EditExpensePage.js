import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expenses.id, expense);
        this.props.history.push('/');
    };

    removeExpense = () => {
        this.props.startRemoveExpense({ id: this.props.expenses.id });
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <div className="page-header">
                 <div className="content-container">
                  <h1 className="page-header__title">Edit Expense</h1>
                 </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                    expense = {this.props.expenses}
                    onSubmit = {this.onSubmit}
                    />
                    <button className="edit-expense__secondary" onClick = {this.removeExpense}>Remove Expense</button>
                </div>
            </div>
        );
    }
}

const connectedEditExpensePage = (state, props) => {
    return{
        expenses: state.expenses.find((expense) => 
        expense.id === props.match.params.id
        )
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(connectedEditExpensePage, mapDispatchToProps)(EditExpensePage);
