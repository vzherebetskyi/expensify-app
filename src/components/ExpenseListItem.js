import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createAt, id }) => (
    <div>
        {/* { expenses.map(element => 
            <p key = {element.description}>{`${element.description}, ${element.amount}, ${element.createAt}`}</p>
            )
        } */}
        <h3><Link to = {`/edit/${id}`}>{description}</Link></h3>
        <p>{amount}, {createAt}</p>
    </div>
);

// const connectedExpenseListItem = (state) => {
//     return {
//         expenses: state.expenses
//     };
// };

export default ExpenseListItem;
// export default ExpenseListItem;