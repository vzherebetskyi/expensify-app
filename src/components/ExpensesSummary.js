import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

import totalExpenses from '../selectors/total-expenses';

export const ExpensesSummary = (props) => {
const { expenses } = props;
 return (
    <div className="page-header">
     <div className="content-container">
        <h1 className="page-header__title">In general there are <span>{expenses.length}</span> expense(-s)! In total you have spent <span>{numeral(totalExpenses(expenses) / 100).format('$0,0.00')}</span></h1>
        <div className="page-header__actions">
            <Link className="expense-summary__add-expense-button" to="/create">Add Expense</Link>
        </div>
     </div>
    </div>
)};

const connectedExpensesSummary = (state) => {
    return {
        expenses: state.expenses
    }
};

export default connect(connectedExpensesSummary)(ExpensesSummary);
