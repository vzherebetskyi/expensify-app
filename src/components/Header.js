import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
         <div>
          <NavLink to = '/' activeClassName = 'is-active' exact = {true}>Go to Dashboard</NavLink>
         </div>
         <div>
          <NavLink to = '/create' activeClassName = 'is-active'>Go to add expense page</NavLink>
         </div>
         <div>
          <NavLink to = '/help' activeClassName = 'is-active'>Go to help page!</NavLink>
         </div>
    </header>
    );

    export default Header;