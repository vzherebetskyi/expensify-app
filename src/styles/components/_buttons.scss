@mixin nice-button {
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;
}

.box-layout__sign-in-button {
    height: $xl-size;
    background: $blue;
    border: none;
    color: white;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
}

.expense-summary__add-expense-button {
    @include nice-button;
    background: $blue;
}

.header__logout-button {
    @include nice-button;
    background: none;
}

.edit-expense__secondary {
    @include nice-button;
    background: #888;
}
