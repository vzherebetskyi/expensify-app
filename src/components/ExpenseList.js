import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = ({expenses}) => (
<div>
    {/* <h1>Expense List</h1> */}
    {
        expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (expenses.map(element => {
                return <ExpenseListItem key = {element.id} {...element} />;
             }))
    }    
</div>
);

const connectedExpenseList = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(connectedExpenseList)(ExpenseList);