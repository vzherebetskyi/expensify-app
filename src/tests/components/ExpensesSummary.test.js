import React from 'react';
import { shallow, mount } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';
import { expenses } from '../test_cases';
import expensesTotal from '../../selectors/total-expenses';

test('should render ExpensesSummary component', () => {
    const wrapper = shallow(<ExpensesSummary expenses = {[]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with one expense', () => {
    const wrapper = mount(<ExpensesSummary expenses = {[expenses[0]]} />);
    expect(wrapper.props().expenses.length).toBe(1);
});

test('should render ExpensesSummary with two expenses', () => {
    const wrapper = mount(<ExpensesSummary expenses = {[expenses[0], expenses[1]]} />);
    expect(wrapper.props().expenses.length).toBe(2);
});

test('should render ExpensesSummary with two expenses in total', () => {
    const wrapper = mount(<ExpensesSummary expenses = {[expenses[0], expenses[1]]} />);
    expect(expensesTotal(wrapper.props().expenses)).toBe(51700);
});