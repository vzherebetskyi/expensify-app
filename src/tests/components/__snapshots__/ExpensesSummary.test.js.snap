// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      In general there are 
      <span>
        0
      </span>
       expense(-s)! In total you have spent 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="expense-summary__add-expense-button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
